File: data/EXP01/golang_no_system_var/2594/2594.output_2.go
func repairCars(ranks []int, cars int) int {
    sort.Sort(sort.Reverse(sort.IntSlice(ranks)))

    totalTime := 0
    totalCarsRepaired := 0

    for _, rank := range ranks {
        carsRepaired := min(cars, rank)
        time := rank * carsRepaired * carsRepaired
        totalTime += time
        totalCarsRepaired += carsRepaired
        cars -= carsRepaired

        if cars == 0 {
            break
        }
    }

    return totalTime
}

func min(a, b int) int {
    if a < b {
        return a
    }
    return b
}

File: data/EXP01/python/2594/2594.output_2.py
# @lc app=leetcode id=2594 lang=python3
from typing import List

class Solution:
    def repairCars(self, ranks: List[int], cars: int) -> int:
        ranks.sort(reverse=True)  # Sort ranks array in descending order
        
        time_taken = 0
        for rank in ranks:
            num_cars = min(cars, rank)  # Assign as many cars as possible to the mechanic
            time_taken += rank * num_cars * num_cars
            cars -= num_cars  # Reduce the number of cars remaining
            
            if cars == 0:
                break  # All cars have been assigned
        
        return time_taken